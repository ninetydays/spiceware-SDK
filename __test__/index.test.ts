import { Polly } from "@pollyjs/core";
import Adapter from "@pollyjs/adapter-node-http";
import Persister from "@pollyjs/persister-fs";
Polly.register(Adapter);
Polly.register(Persister);

import SpicewareSDK from "../src";

const NAME = "SpicewareSDK";
const baseURL = "http://enc-dev.kross.kr:4826";

describe(NAME, () => {
  let polly: Polly;
  let sdk: SpicewareSDK;
  let encryptedName: string;
  const name = "jonghunyu";

  beforeAll(() => {
    polly = new Polly(NAME, {
      adapters: ["node-http"],
      persister: "fs",
    });
    sdk = new SpicewareSDK({ baseURL });
  });

  afterAll(() => polly.stop());
  it("encrypts plain text", async () => {
    const payload = {
      uri: "/test",
      ip: "127.0.0.1",
      fields: { name },
    };
    const { status, data } = await sdk.encrypt(payload);
    expect(status).toBe(200);
    expect(data.result).toBe("success");
    expect(data.fields).toMatchSnapshot();
    encryptedName = data.fields.name;
  });

  it("decrypt encrypted text", async () => {
    const payload = {
      uri: "/test",
      ip: "127.0.0.1",
      fields: { name: encryptedName },
    };
    const { status, data } = await sdk.decrypt(payload);
    expect(status).toBe(200);
    expect(data.result).toBe("success");
    expect(data.fields).toMatchSnapshot();
    expect(data.fields.name).toBe(name);
  });

  it("decrypt encrypted text in array", async () => {
    const payload = {
      uri: "/test",
      ip: "127.0.0.1",
      fields: [{ seq: "1", name: encryptedName }],
    };
    const { status, data } = await sdk.decryptList(payload);
    expect(status).toBe(200);
    expect(data.result).toBe("success");
    expect(data.fields).toMatchSnapshot();
    expect(data.fields[0].name).toBe(name);
  });
});
