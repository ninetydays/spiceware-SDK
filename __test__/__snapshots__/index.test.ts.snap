// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpicewareSDK decrypt encrypted text 1`] = `
Object {
  "name": "jonghunyu",
}
`;

exports[`SpicewareSDK decrypt encrypted text in array 1`] = `
Array [
  Object {
    "name": "jonghunyu",
    "seq": "1",
  },
]
`;

exports[`SpicewareSDK encrypts plain text 1`] = `
Object {
  "name": "hNc2LIumSM8p1uZD5WAeyC8O0gSwLblmkAGWRyt2FzKTS========E",
}
`;
